JSDemos/Demos/Charts/PieWithCustomStyles/Angular/app/app.component.ts

```diff
@@ -0,0 +1,150 @@
+import { NgModule, Component, enableProdMode } from '@angular/core';
+import { BrowserModule } from '@angular/platform-browser';
+import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
+import { DxPieChartModule } from 'devextreme-angular';
+import { registerGradient, registerPattern } from 'devextreme/common/charts';
+import { Data, Service } from './app.service';
+
+if (!/localhost/.test(document.location.host)) {
+  enableProdMode();
+}
+
+@Component({
+  selector: 'demo-app',
+  templateUrl: 'app/app.component.html',
+  styleUrls: ['app/app.component.css'],
+  providers: [Service],
+})
+
+export class AppComponent {
+  data: Data[];
+
+  imagePatternSize = 12;
+
+  shapePatternSize = 6;
+
+  constructor(service: Service) {
+    this.data = service.getData();
+  }
+
+  customizePoint = (point) => {
+    const color = point.series.getPointsByArg(point.argument)[0].getColor();
+    let fillId;
+    switch (point.argument) {
+      case 'Stroke Pattern':
+        fillId = this.getStrokePattern(color);
+        break;
+      case 'Square Pattern':
+        fillId = this.getSquarePattern(color);
+        break;
+      case 'Linear Gradient':
+        fillId = this.getLinearGradient(color);
+        break;
+      case 'Radial Gradient':
+        fillId = this.getRadialGradient(color);
+        break;
+      case 'Pattern with image':
+        fillId = this.getPatternImage(color);
+        break;
+      default:
+        break;
+    }
+
+    return { color: { fillId } };
+  };
+
+  customizeText = (info) => info.argument;
+
+  private hexToRgb = (hex: string, opacity = 1) => {
+    const hexArray = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex);
+    return `rgba(${parseInt(hexArray[1], 16)}, ${parseInt(hexArray[2], 16)}, ${parseInt(hexArray[3], 16)}, ${opacity})`;
+  };
+
+  private gradient = (type: string, color1: string, color2: string) => registerGradient(type, {
+    colors: [{
+      offset: '20%',
+      color: color1,
+    }, {
+      offset: '90%',
+      color: color2,
+    }],
+  });
+
+  private getLinearGradient = (color: string) => this.gradient('linear', color, this.hexToRgb(color, 0.5));
+
+  private getRadialGradient = (color: string) => this.gradient('radial', this.hexToRgb(color, 0.5), color);
+
+  private getPatternImage = (color: string) => registerPattern({
+    width: this.imagePatternSize,
+    height: this.imagePatternSize,
+    template: (container) => {
+      const rect = this.createRect(this.imagePatternSize, color, '', 0);
+      const image = document.createElementNS('http://www.w3.org/2000/svg', 'image');
+      image.setAttribute('x', '0');
+      image.setAttribute('y', '0');
+      image.setAttribute('width', this.imagePatternSize.toString());
+      image.setAttribute('height', this.imagePatternSize.toString());
+      image.setAttribute('href', '../../../../images/Charts/PieWithCustomStyles/diamond.png');
```

Comment:
As far as i know, designer recommends not using of a `png` in demos. Did u agreed this?